request_recorder.go

```diff
@@ -54,8 +70,15 @@ func (r *requestRecorder) String() string {
 		}
 		return str
 	}
-	accepted := requests2str(r.acceptedRequests)
-	unmatched := requests2str(r.unmatchedRequests)
+
+	r.mtx.RLock()
+	defer r.mtx.RUnlock()
+
+	acceptedRequests := r.acceptedRequests
+	unmatchedRequests := r.unmatchedRequests
+
+	accepted := requests2str(acceptedRequests)
+	unmatched := requests2str(unmatchedRequests)
```

Comment:
First two assignments are redundant.
```suggestion
	accepted := requests2str(r.acceptedRequests)
	unmatched := requests2str(r.unmatchedRequests)
```

Reply:
Applied 👍